homework/index.js

```diff
@@ -30,13 +30,162 @@
     return elem;
   }
 
+  function defaultSelectedRepository(data, selectedText) {
+    let elem;
+    data.filter(element => {
+      if (element.name === selectedText) {
+        elem = element;
+      }
+      return elem;
+    });
+    return elem;
+  }
+
+  function buildHtml(data, root) {
+    const header = createAndAppend('header', root, { class: 'header' });
+    createAndAppend('p', header, { text: 'HYF Repositories' });
+    const select = createAndAppend('select', header, {
+      class: 'repo-selector',
+      arialabel: 'HYF Repositories',
+    });
+
+    data
+      .map(repository => repository.name)
+      .sort((a, b) => a.localeCompare(b, { caseFirst: 'lower' }))
+      .map((element, index) => createAndAppend('option', select, { value: index, text: element }));
+
+    const container = createAndAppend('div', root, { id: 'container' });
+    const leftDiv = createAndAppend('div', container, { class: 'left-div whiteframe' });
+    const table = createAndAppend('table', leftDiv);
+    const tbody = createAndAppend('tbody', table, { id: 'tbody' });
+
+    const defaultRepo = defaultSelectedRepository(data, select[select.selectedIndex].text);
+
+    function showSelectedConts(dataCont) {
+      const rightDiv = createAndAppend('div', container, {
+        id: 'rightDiv',
+        class: 'right-div whiteframe',
+      });
+      createAndAppend('p', rightDiv, {
+        class: 'contributor-header',
+        text: 'Contributions :',
+      });
+      const ul = createAndAppend('ul', rightDiv, { id: 'ulList', class: 'contributor-list' });
+      dataCont.forEach(element => {
+        const li = createAndAppend('li', ul, {
+          class: 'contributor-item',
+          arialabel: element.login,
+          tabindex: 0,
+        });
+        createAndAppend('img', li, {
+          src: element.avatar_url,
+          height: 48,
+          class: 'contributor-avatar',
+        });
+        const contInfoDiv = createAndAppend('div', li, { class: 'contributor-data' });
+        createAndAppend('div', contInfoDiv, { text: element.login });
+        createAndAppend('div', contInfoDiv, {
+          class: 'contributor-badge',
+          text: element.contributions,
+        });
+        li.addEventListener('click', () => {
+          window.open(element.html_url, '_blank');
+        });
+      });
+    }
+    fetchJSON(defaultRepo.contributors_url, (err, contributors) => {
+      if (err) {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      } else {
+        showSelectedConts(contributors);
+      }
+    });
+
+    function showSelectedRepo(selectedRepo) {
```

Comment:
Place this function outside of the `buildHtml` function and pass it the `tbody` value that it needs through the parameter list:

```js
function showSelectedRepo(selectedRepo, tbody) {
  ...
}

function buildHtml(data, root) {
  ...
  showSelectedRepo(defaultRepo, tbody);
  ...
}

```